src/main/java/frc/robot/subsystems/drive/Drive.java

```diff
@@ -157,18 +160,53 @@ public void arcadeDrive(double throttle, double turnModifier){
     differentialDrive.arcadeDrive(throttle, turnModifier);
   }
 
+  /**
+   * Executes curvatureDrive with closed loop control. Needs to be called
+   * periodically by the command to update the closed loop. 
+   * 
+   * The closed loop works on the difference between the desired curvature
+   * and the measured curvature, using proportional gain only.
+   * 
+   * @param throttle - [-1.0, 1.0]. Positive is forward
+   * @param curvature - [-1.0, 1.0]. Positive is turning right
+   */
   public void curvatureDrive(double throttle, double curvature) {
-    if(throttle != 0.0 && curvature == 0.0 && !gyroDisabled) { //driving straight and no turn
-      if(this.desiredAngle == Integer.MAX_VALUE) { //means robot just started driving straight
-        this.desiredAngle = gyroscope.getGyroAngle(); 
-      }
-      curvature = this.getAngularError(desiredAngle) * P_VALUE; 
-      this.setLeftPower(throttle - curvature);
-      this.setRightPower(throttle + curvature);
-    }
-    else { // when robot isn't driving straight
-      this.desiredAngle = Integer.MAX_VALUE; //if turn is greater than 0 or if robot is still
-      differentialDrive.curvatureDrive(-throttle, curvature, (Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE));
+
+    if(Math.abs(throttle) < QUICK_TURN_THROTTLE_DEADZONE){
+      // If throttle is less than the deadzone, treat inputs as arcade
+      this.arcadeDrive(throttle, curvature);
```

Comment:
I think we need a disable closed loop condition that goes to regular arcade drive OR curvature drive (both provided by WPILib)

Reply:
If we decide this change is the way to go, all we need to do is add another boolean 
https://github.com/PearceRobotics/rapid-react-2022/pull/49/files#r807454105